cli/cli.js

```diff
@@ -1,102 +1,193 @@
-    default: defaults.FROM_BLOCK,
-  })
-  .option("rpc-url", {
-    alias: ["fork-url", "f"],
-    type: "string",
-    description:
-      "RPC URL - Falls back to `ETH_RPC_URL` env var, then to a public provider",
-    default: process.env.ETH_RPC_URL || defaults.ETH_RPC_URL,
-  })
-  .help(true)
-  .alias("help", "h").argv;
-
-const tableConfig = {
-  columns: {
-    0: { width: 21, wrapWord: true },
-    1: { width: 33, wrapWord: true },
-    2: { width: 21, wrapWord: true },
-    3: { width: 33, wrapWord: true },
-    4: { width: 10, wrapWord: true },
-    5: { width: 10, wrapWord: true },
-    6: { width: 10, wrapWord: true },
-  },
-};
-
-function createTable(events) {
-  let tableData = events.map((event) => [
-    event.guy,
-    event.hash,
-    event.usr,
-    event.tag,
-    event.fax,
-    event.eta,
-    event.status,
-  ]);
-
-  tableData.sort((a, b) => {
-    const etaA = BigInt(a[5]);
-    const etaB = BigInt(b[5]);
-    return etaB > etaA ? 1 : etaB < etaA ? -1 : 0;
-  });
-
-  if (tableData.length === 0) {
-    return "No records to display.";
-  }
-
-  tableData.unshift(["SPELL", "HASH", "USR", "TAG", "FAX", "ETA", "STATUS"]);
-
-  return table(tableData, tableConfig);
+    const spinner = ttyOnlySpinner().start("Fetching pause plans...");
+
+    try {
+        const pause = new ethers.Contract(
+            pauseAddress,
+            defaults.MCD_PAUSE_ABI,
+            ethers.getDefaultProvider(rpcUrl),
+        );
+
+        const plans = await fetchPausePlans(pause, { fromBlock, status });
+        spinner.success("Done!");
+
+        if (format == "TABLE") {
+            console.log(createTable(plans));
+        } else {
+            console.log(createJson(plans));
+        }
+    } catch (error) {
+        spinner.error("Failed!");
+        console.error(chalk.red("An error occurred:", error));
+        process.exit(1);
+    }
 }
 
-async function main() {
-  try {
-    if (argv.rpcUrl == "mainnet") {
-      console.warn(
-        "Falling back to a public provider. For best experience set a custom RPC URL.",
-      );
+/**
+ * Creates a spinner that only shows if stdout is a TTY
+ * @param {...any} args
+ * @returns {import("yocto-spinner").Spinner}
+ */
+function ttyOnlySpinner(...args) {
+    // Only show a spinner if stdout is a TTY
+    if (process.stdout.isTTY) {
+        return yoctoSpinner(...args);
     }
 
-    const pause = new ethers.Contract(
-      argv.pauseAddress,
-      defaults.MCD_PAUSE_ABI,
-      ethers.getDefaultProvider(argv.rpcUrl),
-    );
+    // If not a TTY, return a dummy spinner with empty chainable methods
+    return {
+        start() {
+            return this;
+        },
+        success() {
+            return this;
+        },
+        error() {
+            return this;
+        },
+    };
+}
+
+/**
+ * Converts a list of pause plans to a formatted table
+ * @param {import("./fetchPausePlans").PausePlan[]} plans
+ * @returns {string}
+ */
+function createTable(plans) {
```

Comment:
I know this isn't from this PR specifically, but this function feels a bit out of place in the file. As I see it, it's not a "cli" specific function, but rather a data processing/formatting one.

Same with the tableConfig and createJSON

Reply:
The goal here is to not bloat the directory with multiple files.
Those functions only make sense for the CLI. I don't see any other use for `createTable` and `createJSON` in the context of this package.
We could potentially extract them into different files, but it feels a bit pointless, since they'd only be imported in 1 place.